src/current/v23.1/sso-sql.md

```diff
@@ -1,158 +1,256 @@
+    {{site.data.alerts.end}}
+    
+    {{site.data.alerts.callout_success}}
+    {{ site.data.products.db }}'s IdP configuration can be viewed publicly at: [`https://cockroachlabs.cloud/.well-known/openid-configuration`](https://cockroachlabs.cloud/.well-known/openid-configuration).
+    The link specified for `jwks_uri` provides the IdP's public signing key. For {{ site.data.products.db }} this is `https://cockroachlabs.cloud/oauth2/keys`
+    {{site.data.alerts.end}}
+
+    For example:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    SET CLUSTER SETTING server.jwt_authentication.jwks = '{
+      "keys": [
+        {
+          "alg": "RS256",
+          "kty": "RSA",
+          "kid": "85ba9313fd7a7d4afa84884abcc8403004363180",
+          "e": "AQAB",
+          "use": "sig",
+          "n": "pP-rCe4jkKX6mq8yP1GcBZcxJzmxKWicHHor1S3Q49u6Oe-bQsk5NsK5mdR7Y7liGV9n0ikXSM42dYKQdxbhKA-7--fFon5isJoHr4fIwL2CCwVm5QWlK37q6PiH2_F1M0hRorHfkCb4nI56ZvfygvuOH4LIS82OzIgmsYbeEfwDRpeMSxWKwlpa3pX3GZ6jG7FgzJGBvmBkagpgsa2JZdyU4gEGMOkHdSzi5Ii-6RGfFLhhI1OMxC9P2JaU5yjMN2pikfFIq_dbpm75yNUGpWJNVywtrlNvvJfA74UMN_lVCAaSR0A03BUMg6ljB65gFllpKF224uWBA8tpjngwKQ"
+        },
+        {
+          "alg": "RS256",
+          "n": "8ASqcDDTYmFx53uAkWlmqx8qIx0WQacaswAAb7xGx9XL2T71VPtKKgPqoimXzj4fXT2F3opIgOQgGKxkxw2QrAAoejdIud5URrZLRponmbMQO3erG5sd8PC29rjekX8_hvX_AMT3zU8JLKVLFR0xJmFdmIomnarpEvWKmG0SNvbuVveprsve0n1W35uodcRe417vlTNnH9j8fesQekvFf8tIWbHouXDg3B1km4gqZBbQ_MWvGriGBY5sfr7A2d0iNe89Aje7pz1RFLUFOu-u_NyD6RwL6qo4_yetAYIzm02a2KAAq03YPs2LHMVQmkh1LtyeuA6bvf9146cAFY4BCQ",
+          "use": "sig",
+          "e": "AQAB",
+          "kty": "RSA",
+          "kid": "05150a1320b9395b05716877376928509bab44ac"
+        }
+      ]
+    }';
+    ~~~
+
+1. Configure token generation.
 
 {{site.data.alerts.callout_success}}
-CockroachDB {{ site.data.products.cloud }}'s IdP configuration can be viewed publicly at: [`https://cockroachlabs.cloud/.well-known/openid-configuration`](https://cockroachlabs.cloud/.well-known/openid-configuration).
-The link specified for `jwks_uri` provides the IdP's public signing key. For CockroachDB {{ site.data.products.cloud }} this is `https://cockroachlabs.cloud/oauth2/keys`
+In order for this feature to work DB Console SSO and cluster SSO must both be configured with the same IDP.
 {{site.data.alerts.end}}
 
-For example:
+The `use_token` field determines which part of the received OIDC credentials the cluster uses to [determine the user’s identity](#how-cockroachdb-determines-the-sql-username-from-a-jwt).
+
+It can be set to either `id_token` or `access_token`. Here we set it to `id_token` because ...
+
+In another examples where ... we would set it to `access_token`. Okta I'm guessing?
```

Comment:
@kpatron-cockroachlabs

Reply:
I just don't know this stuff. please provide the knowledge, kind sir

Reply:
It just has to do with the mechanics of how the IDP generates tokens. Sometimes the access tokens are easier to map and contain the information you want, sometimes it is the access token. It just depends on the IDP and how you want to configure it.

Reply:
How does one tell which to use in a particular case. When I was setting it up for GCP, I only knew because @matthewtodd told me. Should we just list a few use cases, like google, okta, etc.,? This is ultimately about whether you ultimately want to use email or ID to identify the user via the identity mapping, correct? just kind of spelling out the path the info?

Reply:
how do you like... inspect a token to check? sorry if that's a dumb question
just trying to think through things from the user perspective, and that seems the most important question: how would I figure it out if I didn't already know and was trying to set this up?

Reply:
Basically per IDP it might not contain the right fields (email, audience, etc) on one token or the other. It is very dependent on how you configure your IDP. Basically you either need to read your IDP docs, try one then the other, or inspect the token. Once you have a token you can base 64 decode the middle bit (between the two periods) to see what information is included. This is our internal link that may be helpful https://cockroachlabs.atlassian.net/wiki/spaces/MC/pages/2825683043/Debugging+Cluster+SSO#Checking-the-contents-of-a-token

Reply:
ok, thank you, hopefully this is helpful enough